iroha-cli/interactive/impl/interactive_query_cli.cpp

```diff
@@ -169,45 +170,65 @@ namespace iroha_cli {
     }
 
     nonstd::optional<iroha::model::Pager>
-    InteractiveQueryCli::parsePager(const QueryParams& params) {
+    InteractiveQueryCli::parsePager(QueryParams params) {
       const auto encoded_tx_hash = params[0];
       const auto limit_str = params[1];
+      /**
+       * Use '-' as the argument of tx_hash to specify that the hash is empty
+       * because interactive cli splits arguments by space.
+       */
       const auto PAGER_TX_HASH_EMPTY = "-";
 
       Pager pager;
       if (encoded_tx_hash == PAGER_TX_HASH_EMPTY) {
         pager.tx_hash.fill(0);
       } else {
-        const auto decoded_hash = iroha::hexstringToBytestring(encoded_tx_hash);
+        const auto decoded_hash =
+          iroha::hexstringToBytestring(encoded_tx_hash);
         if (not decoded_hash) {
           return nonstd::nullopt;
         }
         pager.tx_hash = iroha::hash256_t::from_string(*decoded_hash);
       }
 
-      try {
-        pager.limit = static_cast<uint16_t>(std::stoul(limit_str));
-      } catch(...) {
+      pager.limit = [&pager, &limit_str] {
```

Comment:
Not really the way I wanted, look at this:
```cpp
return nonstd::make_optional<..>
    | [](auto pager) {
      if (encoded_tx_hash == PAGER_TX_HASH_EMPTY) {
        pager.tx_hash.fill(0);
      } else {
        const auto decoded_hash =
          iroha::hexstringToBytestring(encoded_tx_hash);
        if (not decoded_hash) {
          return nonstd::nullopt;
        }
        pager.tx_hash = iroha::hash256_t::from_string(*decoded_hash);
      }
      return pager;
    } | [&limit_str](auto pager) {
        try {
          pager.limit = boost::lexical_cast<decltype(pager.limit)>(limit_str);
        } catch (...) {
          return nonstd::nullopt;
        }
        return pager;
    }
```

Looks pretty same but with monadic binds. Choose the previous way or this one, thanks